stdlib/src/builtin/string_literal.mojo

```diff
@@ -287,6 +287,18 @@ struct StringLiteral(
             unsafe_pointer=self.unsafe_ptr(), length=self.byte_length()
         )
 
+    fn __reversed__(
+        ref [_]self,
```

Comment:
I think `self` can be a borrowed argument.

Reply:
Both `String` and `StringLiteral` `__iter__` funcs have `ref [_]self` as signature (String has it's `__reversed__` func the same)

Reply:
There, removed them and nothing exploded :D

Reply:
I think `StringLiteral.__iter__` can use `borrowed self` since SL doesn't support mutable iterations.